frontend/src/app/core/services/confirm-window.service.ts

```diff
@@ -0,0 +1,93 @@
+import { ComponentFactoryResolver, ComponentRef, Injectable } from '@angular/core';
+import { ButtonOptions } from '@shared/models/button-options';
+import { ConfirmationService, PrimeIcons } from 'primeng/api';
+
+import { ConfirmWindowComponent } from '../components/confirm-window/confirm-window.component';
+import { HelperService } from './helper.service';
+
+const DEFAULT_VISIBLE = true;
+const DEFAULT_ACCEPT = 'Accept';
+const DEFAULT_CANCEL = 'Cancel';
+const DEFAULT_CLASS = 'p-button-primary';
+const DEFAULT_ICON = PrimeIcons.INFO_CIRCLE;
+
+@Injectable({
+    providedIn: 'root',
+})
+export class ConfirmWindowService {
+    confirmWindowRef: ComponentRef<ConfirmWindowComponent>;
+
+    constructor(
+        private confirmationService: ConfirmationService,
+        private helper: HelperService,
+        private componentFactoryResolver: ComponentFactoryResolver
+    ) { }
+
+    /**Creates confirmation window
+     * @param title title of the window
+     * @param message message of window
+     * @param icon default: INFO_CIRCLE, icon before message text
+     * @param isClosable default: false, can be closed by esc or close mark
+     * @param acceptButton default: visible, accept button options
+     * @param cancelButton default: visible, cancel button options
+     * @param accept function, that invokes after accept button click
+     * @param cancel function, that invokes after cancel button click
+     */
+    confirm(options: {
+        title: string,
+        message: string,
+        icon?: string,
+        isClosable?: boolean,
+        accept?: Function,
```

Comment:
Move functions to the end, and rewrite them to method signature syntax type of function declaration